miq_bisect_gemfile_parser/bisect_gemfile_parser

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env ruby
+
+# This is a helper for running a bisect when having to deal with "point in
+# time" issues when bisecting and dealing with git-gems
+#
+# Because our Gemfile use the most recent version of the gem when running a
+# `bundle` and/or `bundle update`, when attempting to do a `git bistect`,
+# dependecies that might be causing the issue are not being checked out in a
+# fashion that represents what the state of things were in the MIQ environment
+# at the point of the checkout.
+#
+# This helper attempts to checkout a proper revision of all of the manageiq git
+# gem dependencies by checking the current revision of the gems in the
+# currently parse Gemfile, and throws a override file in the `bundler.d/`
+# directory which will set the proper overrides.
+#
+# The workflow for using this script is as follows;
+#
+#   * Start `git bisect`
+#   * Run the script
+#     - you can actually chuck this in `bin/` since it is git-ignored)
+#   * Run `bin/bundle update`
+#   * Do what you need to validate the git sha, and move on the the next commit
+#   * Repeat steps 2-4 until you are finished
+#   * Run `rm bundle.d/bisect.rb` to clean up this script (DON'T FORGET!)
+#
+# The last step is important as it will most likely mess with your your master
+# branch.
+#
+
+class GemfilePluginParser
+  GIT_URL_MATCH = /^\w+:\/\/github.com\/ManageIQ\/(manageiq-|miq).*$/.freeze
+
+  class << self
+    def parse
+      new.tap {|i| i.instance_eval { eval File.read "Gemfile" } }
+         .manageiq_plugins
+    end
+  end
+
+  attr_reader :manageiq_plugins
+
+  def initialize
+    @manageiq_plugins = []
+  end
+
+  def gem gem_name, *args
+    hash = args.pop
+    if hash and hash.is_a?(Hash) and hash[:git].to_s =~ GIT_URL_MATCH
+      @manageiq_plugins << gem_name
+    end
+  end
+
+  def __dir__
+    Dir.pwd
+  end
+  def source *args; end;
+  def eval_gemfile *args; end;
+  def group *args
+    yield if block_given?
+  end
+
+  def dependencies
+    []
+  end
+end
+
+base_bisect_rb_content = [
+  'require "shellwords"',
```

Comment:
would this work as here doc and a `split("\n")`? or a `%{...}`?

Reply:
Probably.  Unsure why I went the `Array` route to be honest.  Will take a look at some point and get back.

Reply:
Oh, just took a look.  Because there isn't a `non-interpolation` mode for heredoc strings in ruby, I went with a collection of single quote strings.  See the last line as to why.